docs/source/templates/class_api.rst

```diff
@@ -0,0 +1,197 @@
+..
+  Use the following convention for headings:
+
+    # with overline, for parts
+
+    * with overline, for chapters
+
+    = for sections
+
+    - for subsections
+
+    ^ for subsubsections
+
+    " for paragraphs
+
+..
+  Class / method / container name)
+  for free functions that are callable, preserve the naming convention, `view_alloc()`
+
+``{{ entity_name }}``
+=====================
+
+.. role:: cppkokkos(code)
+    :language: cppkokkos
+
+..
+  The (pulic header) file the user will include in their code
+
+Header File: ``{{ core_header_file }}``
+
+..
+  High-level, human-language summary of what the thing does, and if possible, brief statement about why it exists (2 - 3 sentences, max);
+
+Description
+-----------
+
+..
+  The API of the entity.
+
+Interface
+---------
+
+..
+  The declaration or signature of the entity.
+
+.. cppkokkos:class:: template <class DataType, class... Traits> CoolerView
+
+  ..
+    Template parameters (if applicable)
+    Omit template parameters that are just used for specialization/is deduced/ and/or should not be exposed to the user.
+
+  .. rubric:: Template Parameters
+
+  :tparam Foo: Description of the Foo template parameter
+
+  ..
+    Parameters (if applicable)
+
+  .. rubric:: Parameters
+
+  :param bar: Description of the bar parameter
+
+  .. rubric:: Public Types
+
+  .. cppkokkos:type:: data_type
+
+    Some interesting description of the type and how to use it.
+
+  .. rubric:: Static Public Member Variables
+
+  .. cppkokkos:member:: int some_var = 5;
+
+    Description of some_var
+
+    ..
+      If you have related info
+
+    .. seealso::
+
+      ..
+        We can cross-reference entities
+
+      The :cppkokkos:func:`frobrnicator` free function.
+
+  .. rubric:: Constructor
+
+  .. cppkokkos:function:: CoolerView(CoolerView&& rhs)
+
+    Whether it's a move/copy/default constructor. Describe what it does.
+
+  ..
+    Only include the destructor if it does something interesting as part of the API, such as RAII classes that release a resource on their destructor. Classes that merely
+    clean up or destroy their members don't need this member documented.
+
+  .. rubric:: Destructor
+
+  .. cppkokkos:function:: ~CoolerView()
+
+    Document what special effect the destructor has.
+
+  .. rubric:: Public Member Functions
+
+  .. cppkokkos:function:: template<class U> foo(U x)
+
+    Brief description of the function.
+
+    :tparam U: Description of U
+
+    :param: description of x
+
+    ..
+      Describe any API changes between versions.
+
+    .. versionchanged:: 3.7.1
+
+      Only takes one parameter for foo-style operations instead of two.
+
+  ..
+    Deprecated types need special handling, both for the badge displayed beforehand and the deprecated version directive.
+    Note, there is not currently any functionality for other deprecated entities.
+
+  .. cppkokkos:deprecated-type:: 4.0.1 foobar
```

Comment:
In your comment, maybe note key points of our discussion this morning about deprecation syntax in .rst files, and (parser) rendering in html.

Reply:
Done! Though I did move to the c++ style `[[deprecated("reason")]]`